beacon_node/network/src/sync/network_context.rs

```diff
@@ -281,6 +365,46 @@ impl<T: BeaconChainTypes> SyncNetworkContext<T> {
                     .insert(id, (batch_id, block_blob_info));
                 Ok(id)
             }
+            ByRangeRequestType::BlocksAndDataColumns => {
+                debug!(
+                    self.log,
+                    "Sending backfill BlocksByRange and DataColumnsByRange requests";
+                    "method" => "Mixed by range request",
+                    "count" => request.count(),
+                    "peer" => %peer_id,
+                );
+
+                // create the shared request id. This is fine since the rpc handles substream ids.
+                let id = self.next_id();
+                let request_id = RequestId::Sync(SyncRequestId::BackFillBlockAndDataColumns { id });
+                let data_column_ids = data_column_ids.ok_or(
+                    "Trying to send a backfill BlocksByRange and DataColumnsByRange request without data column ids"
+                )?;
+
+                // Create the blob request based on the blob request.
+                let data_columns_request = Request::DataColumnsByRange(DataColumnsByRangeRequest {
+                    start_slot: *request.start_slot(),
+                    count: *request.count(),
+                    data_column_ids: data_column_ids,
+                });
+                let blocks_request = Request::BlocksByRange(request);
+
+                // Send both requests. Make sure both can be sent.
+                self.send_network_msg(NetworkMessage::SendRequest {
+                    peer_id,
+                    request: blocks_request,
+                    request_id,
+                })?;
+                self.send_network_msg(NetworkMessage::SendRequest {
+                    peer_id,
```

Comment:
Is this peer a custodian of the requested columns?

Reply:
it should be yes, will add a `TODO` here mentioning that we need to implement logic to choose which peers to send the requests to